### ISSUE TITLE
Short Weierstrass point compression

### ISSUE THREAD

**Issue by carrion256** [lambdaclass/lambdaworks#684]
I've noticed a [TODO](https://github.com/lambdaclass/lambdaworks/blob/main/math/src/elliptic_curve/short_weierstrass/point.rs#L237) in the codebase but couldn't find a ticket for it. 

I'd made a partial implementation [here](https://github.com/near/rollup-data-availability/pull/53/commits/3d3bb10c895e20a504653475e4deffd6ce26f885) but since removed it because I've only recently been diving deeper into ECC and there's not a huge requirement for us right now.

Relates to https://github.com/lambdaclass/lambdaworks/issues/323

**Comment by MauroToscano** [comment]
Yes. A first step is to just convert them to affine, and you can save 1 field element. Then you could also add a compressed format, with x coordinate and a bit to know which of the 2 you have to use.
